### PR TITLE
Promote Replication Controller and Replica Set e2e tests to Conformance

### PR DESCRIPTION
**Pull request by pontiyaraja**
**What this PR does / why we need it**:
[Replication controller] releasing/removing the pods which is no longer matched with the created label. 
[Replication controller] should adopt matching pods on replication controller creation
[Replica Set] should adopt matching pods on replica set creation and release no longer matching pods from replica set. 
Promoting these e2e test to conformance test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
No flaky
**Release note**:

```release-note
NONE
```
/area test
/sig rc
/sig testing
/sig architecture

/release-note-none

cc @spiffxp

/assign @bgrant0607 @janetkuo

### CODE REVIEW

File: test/e2e/apps/rc.go [hunk 1]
```diff
@@ -64,7 +64,13 @@ var _ = SIGDescribe("ReplicationController", func() {
 		testRCAdoptMatchingOrphans(f)
 	})
 
-	It("should release no longer matching pods", func() {
+	/*
+		Release : v1.13
+		Testname: Replication Controller, release pods
+		Description: Replication Controller Should release no longer matching pods, By creating a given Replicatio Controller. Pod should released, by using PollImmediate by checking the label of the pod .
```
Comment (spiffxp) [inline test/e2e/apps/rc.go]: s/Should/MUST/

A Replication Controller is created, and its Pods are created.  When the labels on one of the Pods change to no longer match the Replication Controller's label selector, the Replication Controller MUST release the Pod and update the Pod's owner references.

**Review by spiffxp — COMMENTED** [review]
Note that I'd rather we not focus on workload API's right now.  That said, this looks like a valid conformance test to me, relying on nothing special.  Update the description and assuming this isn't flaky I'll /lgtm

**Comment by spiffxp** [comment]
/unassign @bgrant0607 
not ready for approval yet

File: test/e2e/apps/rc.go (hunk 1 above)
Comment (pontiyaraja) [reply test/e2e/apps/rc.go]: updated test description and added tag

>  [Conformance]

**Comment by pontiyaraja** [comment]
@spiffxp @bgrant0607 @foxish

File: test/e2e/apps/rc.go [hunk 2]
```diff
@@ -67,10 +67,10 @@ var _ = SIGDescribe("ReplicationController", func() {
 	/*
 		Release : v1.13
 		Testname: Replication Controller, release pods
-		Description: Replication Controller Should release no longer matching pods, By creating a given Replicatio Controller. Pod should released, by using PollImmediate by checking the label of the pod .
+		Description: A Replication Controller (RC) is created, and its Pods are created. When the labels on one of the Pods change to no longer match the RC's label selector, the RC MUST release the Pod and update the Pod's owner references
 	*/
 
-	framework.ConformanceIt("should release no longer matching pods", func() {
+	framework.ConformanceIt("should release no longer matching pods [Conformance]", func() {
```
Comment (spiffxp) [inline test/e2e/apps/rc.go]: Please remove the `[Conformance]` tag, this is what `ConformanceIt` does

Comment (pontiyaraja) [reply test/e2e/apps/rc.go]: @spiffxp Removed [Conformance] label

**Comment by spiffxp** [comment]
sorry, last request before I /lgtm, can you squash down into a single commit please

**Comment by spiffxp** [comment]
cleaning up stray labels

/remove-area apiserver
/remove-area kubeadm
/remove-area kubectl
/remove-area kubelet
/remove-sig api-machinery
/remove-sig api-cli
/remove-sig cloud-provider
/remove-sig cluster-lifecycle
/remove-sig gcp
/remove-sig node
/remove-sig release
/remove-sig scheduling
/remove-sig testing

**Comment by spiffxp** [comment]
/remove-sig cli
copy pasta tastes delicious

**Comment by spiffxp** [comment]
/lgtm
/assign @smarterclayton 
for the /approve

File: test/e2e/apps/rc.go [hunk 3]
```diff
@@ -64,7 +64,13 @@ var _ = SIGDescribe("ReplicationController", func() {
 		testRCAdoptMatchingOrphans(f)
 	})
 
-	It("should release no longer matching pods", func() {
+	/*
+		Release : v1.13
+		Testname: Replication Controller, release pods
+		Description: A Replication Controller (RC) is created, and its Pods are created. When the labels on one of the Pods change to no longer match the RC's label selector, the RC MUST release the Pod and update the Pod's owner references
+	*/
+
```
Comment (mgdevstack) [inline test/e2e/apps/rc.go]: It's good if we can remove the line break between comment and code block.

**Comment by bgrant0607** [comment]
Looks like it shouldn't be flaky, since it waits for the ownerReferences update to happen.

The test description could be more specific about what it is doing.

Why change just this ownerReferences-related test and not the others?

@caesarxuchao

**Comment by caesarxuchao** [comment]
Why do we pick the replication controller test instead of the replicaset test? `testRSAdoptMatchingAndReleaseNotMatching` is a similar test for the replicaset.

**Comment by pontiyaraja** [comment]
@caesarxuchao `testRSAdoptMatchingAndReleaseNotMatching` covers both non matching label pod release and adopting orphan pod. I will make change in widely used next generation replication controller test.

**Comment by caesarxuchao** [comment]
There is no faliure for these three tests in the last two weeks in the presubmit runs, so lgtm:

https://storage.googleapis.com/k8s-gubernator/triage/index.html?ci=0&pr=1&job=pull-kubernetes-e2e-gce&test=should%20adopt%20matching%20pods%20on%20creation

https://storage.googleapis.com/k8s-gubernator/triage/index.html?ci=0&pr=1&job=pull-kubernetes-e2e-gce&test=should%20release%20no%20longer%20matching%20pods

https://storage.googleapis.com/k8s-gubernator/triage/index.html?ci=0&pr=1&job=pull-kubernetes-e2e-gce&test=should%20adopt%20matching%20pods%20on%20creation%20and%20release%20no%20longer%20matching%20pods

File: test/e2e/apps/rc.go [hunk 4]
```diff
@@ -60,11 +60,21 @@ var _ = SIGDescribe("ReplicationController", func() {
...
+	*/
+	framework.ConformanceIt("should adopt matching pods on creation", func() {
 		testRCAdoptMatchingOrphans(f)
 	})
 
-	It("should release no longer matching pods", func() {
+	/*
+		Release : v1.13
+		Testname: Replication Controller, release pods
+		Description: Replication Controller MUST release the Pod, when the labels on one of the Pods change to no longer match the RC's label selector, and update the Pod's owner references
```
Comment (spiffxp) [inline test/e2e/apps/rc.go]: A Replication Controller (RC) is created, and its Pods are created. When the labels on one of the Pods change to no longer match the RC's label selector, the RC MUST release the Pod and update the Pod's owner references.

File: test/e2e/apps/rc.go [hunk 5]
```diff
@@ -60,11 +60,21 @@ var _ = SIGDescribe("ReplicationController", func() {
 		testReplicationControllerConditionCheck(f)
 	})
 
-	It("should adopt matching pods on creation", func() {
+	/*
+		Release : v1.13
+		Testname: Replication Controller, adopt matching pods
+		Description: Replication Controller with a matching selector, should adopt matching pods by checking the owner's reference
```
Comment (spiffxp) [inline test/e2e/apps/rc.go]: An ownerless Pod is created, then a Replication Controller (RC) is created whose label selector will match the Pod.  The RC MUST either adopt the Pod or delete and replace it with a new Pod.

File: test/e2e/apps/replica_set.go [hunk 6]
```diff
@@ -103,7 +103,12 @@ var _ = SIGDescribe("ReplicaSet", func() {
 		testReplicaSetConditionCheck(f)
 	})
 
-	It("should adopt matching pods on creation and release no longer matching pods", func() {
+	/*
+		Release : v1.13
+		Testname: Replica Set, adopt matching pods and release non matching pods
+		Description: Orphan pods shoud get adopted by checking the owner's reference by the replica set which is created with a matching selector, and when the matched label of the replica set's pods changed, it shout release those pods by comparing the owners owner's reference.
+	*/
+	framework.ConformanceIt("should adopt matching pods on creation and release no longer matching pods", func() {
```
Comment (spiffxp) [inline test/e2e/apps/replica_set.go]: A Pod is created, then a Replica Set (RS) whose label selector will match the Pod.  The RS MUST either adopt the Pod or delete and replace it with a new Pod.  When the labels on one of the Pods owned by the RS change to no longer match the RS's label selector, the RS MUST release the Pod and update the Pod's owner references

**Review by spiffxp — COMMENTED** [review]
Suggestions for more detailed descriptions

**Comment by spiffxp** [comment]
/lgtm
/assign @bgrant0607 
This is ready for /approve

**Comment by bgrant0607** [comment]
Thanks for adding the RS tests and adoption tests, and for improving the descriptions.

/approve

**Comment by spiffxp** [comment]
We're hitting the TestVolumeProvision flake on integration tests https://github.com/kubernetes/kubernetes/issues/70180

**Pull request merged**